pkgs/development/libraries/tbb/2020_3.nix

```diff
@@ -0,0 +1,104 @@
+      sha256 = "fhorfqO1hHKZ61uq+yTR7eQ8KYdyLwpM3K7WpwJpV74=";
+    })
+
+    # Fixes build with upcoming gcc-13:
+    #  https://github.com/oneapi-src/oneTBB/pull/833
+    (fetchurl {
+      name = "gcc-13.patch";
+      url = "https://github.com/oneapi-src/oneTBB/pull/833/commits/c18342ba667d1f33f5e9a773aa86b091a9694b97.patch";
+      sha256 = "ZUExE3nsW80Z5GPWZnDNuDiHHaD1EF7qNl/G5M+Wcxg=";
+    })
+
+    # Fixes build for aarch64-darwin
+    (fetchurl {
+      name = "aarch64-darwin.patch";
+      url = "https://github.com/oneapi-src/oneTBB/pull/258/commits/86f6dcdc17a8f5ef2382faaef860cfa5243984fe.patch";
+      sha256 = "sha256-JXqrFPCb3q1vfxk752tQu7HhApCB4YH2LoVnGRwmspk=";
+    })
+  ];
+
+  nativeBuildInputs = (lib.optionals stdenv.isDarwin [
+    fixDarwinDylibNames
+  ]);
+
+  makeFlags = lib.optionals stdenv.cc.isClang [
+    "compiler=clang"
+  ] ++ (lib.optional (stdenv.buildPlatform != stdenv.hostPlatform)
+    (if stdenv.hostPlatform.isAarch64 then "arch=arm64"
+    else if stdenv.hostPlatform.isx86_64 then "arch=intel64"
+    else throw "Unsupported cross architecture"));
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib
+    cp "build/"*release*"/"*${stdenv.hostPlatform.extensions.sharedLibrary}* $out/lib/
+    mv include $out/
+    rm $out/include/index.html
+
+    runHook postInstall
+  '';
+
+  postInstall = let
+    pcTemplate = fetchurl {
+      url = "https://github.com/oneapi-src/oneTBB/raw/478de5b1887c928e52f029d706af6ea640a877be/integration/pkg-config/tbb.pc.in";
+      sha256 = "2pCad9txSpNbzac0vp/VY3x7HNySaYkbH3Rx8LK53pI=";
+    };
+  in ''
+    # Generate pkg-config file based on upstream template.
+    # It should not be necessary with tbb after 2021.2.
+    mkdir -p "$out/lib/pkgconfig"
+    substitute "${pcTemplate}" "$out/lib/pkgconfig/tbb.pc" \
+      --subst-var-by CMAKE_INSTALL_PREFIX "$out" \
+      --subst-var-by CMAKE_INSTALL_LIBDIR "lib" \
+      --subst-var-by CMAKE_INSTALL_INCLUDEDIR "include" \
+      --subst-var-by TBB_VERSION "${version}" \
+      --subst-var-by TBB_LIB_NAME "tbb"
+  '';
+
+  meta = with lib; {
+    description = "Intel Thread Building Blocks C++ Library";
+    homepage = "http://threadingbuildingblocks.org/";
+    license = licenses.asl20;
+    longDescription = ''
+      Intel Threading Building Blocks offers a rich and complete approach to
+      expressing parallelism in a C++ program. It is a library that helps you
+      take advantage of multi-core processor performance without having to be a
+      threading expert. Intel TBB is not just a threads-replacement library. It
+      represents a higher-level, task-based parallelism that abstracts platform
+      details and threading mechanisms for scalability and performance.
+    '';
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ thoughtpolice tmarkus ];
```

Comment:
I diffed manually previous version with the current one and got this diff:

```diff
--- default_.nix        2023-03-01 06:26:50.468888240 +0000
+++ 2020_3.nix  2023-03-01 06:27:05.133114732 +0000
@@ -47,9 +47,9 @@
     })
   ];

-  nativeBuildInputs = lib.optionals stdenv.isDarwin [
+  nativeBuildInputs = (lib.optionals stdenv.isDarwin [
     fixDarwinDylibNames
-  ];
+  ]);

   makeFlags = lib.optionals stdenv.cc.isClang [
     "compiler=clang"
@@ -58,8 +58,6 @@
     else if stdenv.hostPlatform.isx86_64 then "arch=intel64"
     else throw "Unsupported cross architecture"));

-  enableParallelBuilding = true;
-
   installPhase = ''
     runHook preInstall

@@ -101,6 +99,6 @@
       details and threading mechanisms for scalability and performance.
     '';
     platforms = platforms.unix;
-    maintainers = with maintainers; [ thoughtpolice ];
+    maintainers = with maintainers; [ thoughtpolice tmarkus ];
   };
 }
```

A few questions:
1. Why is it not just a copy of the previous file? `maintainers` change sounds reasonable (except that it does not specify you, but `tmarkus`, is it intentional?).
2. Disabling build parallelism by dropping `enableParallelBuilding` is unexpected. Can you clarify why you need to disable it? I think it's also the only change that causes many rebuilds to happen.
3. Adding seemingly superfluous braces around `nativeBuildInputs` looks redundant as well.

Reply:
1. i guess the changes are needed
2. this was suggested by @SuperSandro2000 in https://github.com/NixOS/nixpkgs/pull/214762#discussion_r1111990017
3. @hesiod i guess you did that?

Reply:
> this was suggested by @SuperSandro2000 in
https://github.com/NixOS/nixpkgs/pull/214762#discussion_r1111990017

The suggestion makes sense for 2021.8 as it uses `cmake` in it's `nativeBuildInputs`. One of it's non-trivial side-effects is to import `setup-hook.sh` into every builder that uses it: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L128-L131. There it enables parallelism by default:

```bash
    if ! [[ -v enableParallelBuilding ]]; then
        enableParallelBuilding=1
        echo "cmake: enabled parallel building"
    fi
```

But `2020.3` does not use `cmake` and uses plain `make`. I think it's still needed for `2020.3`.

Reply:
Good catch. I was not aware of that.

Reply:
>I think it's also the only change that causes many rebuilds to happen.

yes, now the package does not need to be build according to nixpkgs-review